File: since2020/CodeForces/1327A.py
'''
	@judge CodeForces
	@id 1327A
	@name Sum of Odd Integers
	@contest Educational Codeforces Round 84

	@tag Math
'''
from sys import stdin

def solve(n, k):
	if k * k > n:
		return False
	return n % 2 == k % 2

input()
for line in stdin:
	n, k = map(int, line.split())
	print('YES' if solve(n, k) else 'NO')

File: since2020/CodeForces/1327A.cpp
/**
 * 	@judge CodeForces
 *  @id 1327A
 *  @name Sum of Odd Integers
 *  @contest Educational Codeforces Round 84
 * 
 *  @tag Math
 */
#include <bits/stdc++.h>
using namespace std;

typedef long long int LL;

int n, k;

bool solve() {
	if (n < (LL) k * k)
		return false;
	return (n % 2) ^ (k % 2 == 0);
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	cout.tie(nullptr);
	int t;
	for (cin >> t; t; t--) {
		cin >> n >> k;
		cout << (solve() ? "YES" : "NO") << endl;
	}
}